clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2772,11 +2772,15 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
                                        const CXXRecordDecl *KernelObj) {
     TypeSourceInfo *TSInfo =
         KernelObj->isLambda() ? KernelObj->getLambdaTypeInfo() : nullptr;
-    auto Type = QualType(KernelObj->getTypeForDecl(), 0);
-    Type->getAsRecordDecl()->setAnonymousStructOrUnion(KernelObj->isLambda());
+    // Generate a name kernel object variable.
+    Twine Name = Twine("__SYCLKernel");
+    SmallString<12> Buffer;
+    StringRef NameRef = Name.toStringRef(Buffer);
+    auto &Ident = Ctx.Idents.getOwn(NameRef);
```

Comment:
You only need Ctx.Idents.get("__SYCLKernel");.  That said, I might suggest you do  this ONLY if hte kernelobj identifier is empty.

Additionally:
You can make a string-ref to a constant string (in fact, you can make a llvm::StringLiteral to one), going through the twine and the buffer is unnecessary.

Reply:
> do this ONLY if hte kernelobj identifier is empty.

That's a good point! Thanks. :)